src/sentry/api/endpoints/project_metrics_extraction_rules.py

```diff
@@ -42,15 +40,18 @@ def delete(self, request: Request, project: Project) -> Response:
         if not self.has_feature(project.organization, request):
             return Response(status=404)
 
-        rules_update = request.data.get("metricsExtractionRules") or []
-        if len(rules_update) == 0:
+        config_update = request.data.get("metricsExtractionRules") or []
+        if len(config_update) == 0:
             return Response(status=204)
 
         try:
-            state_update = self._generate_deleted_rule_objects(rules_update)
-            delete_metrics_extraction_rules(project, state_update)
+            with transaction.atomic(router.db_for_write(SpanAttributeExtractionRuleConfig)):
+                for obj in config_update:
+                    SpanAttributeExtractionRuleConfig.objects.filter(
+                        project=project, span_attribute=obj["spanAttribute"]
+                    ).all().delete()
```

Comment:
You do not need `all()` here, just `filter()` is enough
```suggestion
                    SpanAttributeExtractionRuleConfig.objects.filter(
                        project=project, span_attribute=obj["spanAttribute"]
                    ).delete()
```